processing/src/main/java/io/druid/segment/incremental/IncrementalIndex.java

```diff
@@ -512,26 +516,28 @@ private long getMaxTimeMillis()
     return getFacts().lastKey().getTimestamp();
   }
 
-  private String[] getDimVals(final DimDim dimLookup, final List<String> dimValues)
+  private int[] getDimVals(final DimDim dimLookup, final List<String> dimValues)
   {
-    final String[] retVal = new String[dimValues.size()];
     if (dimValues.size() == 0) {
       // NULL VALUE
-      if (!dimLookup.contains(null)) {
-        dimLookup.add(null);
-      }
+      dimLookup.add("");
```

Comment:
why the behavior change? earlier it was dimLookup.add(null) and now changed to dimLookup.add("") .